diagnostics/TUTT/TUTT_calc_model.py

```diff
@@ -0,0 +1,648 @@
+#Jun 29 2021
+import matplotlib.pyplot as pyplot
+#from mpl_toolkits.basemap import Basemap
+#from mpl_toolkits.basemap import addcyclic
+import numpy as np
+from normalize import normalize
+# http://netcdf4-python.googlecode.com/svn/trunk/docs/netCDF4-module.html
+from netCDF4 import Dataset, num2date
+# http://scikit-image.org/
+from skimage import measure
+# conda install -c scitools shapely
+from shapely.geometry import Polygon, LineString
+from scipy.signal import argrelextrema
+from scipy.stats.stats import pearsonr
+from scipy.interpolate import interp2d
+from scipy.interpolate import RectBivariateSpline
+import cartopy.crs as ccrs
+import cartopy.mpl.ticker as cticker
+
+lat_coord = os.environ["lat_coord"]
+lon_coord = os.environ["lon_coord"]
+z200_var = os.environ["z200_var"]
+WK_DIR    = os.environ["WK_DIR"]
+CASENAME  = os.environ["CASENAME"]
+
+def add_cyclic_BL(dat,lon):
+  if dat.shape[2]!=lon.shape[0]:
+    print('Dimension error in the input data!')
+    return
+  dat1=np.zeros((dat.shape[0],dat.shape[1],dat.shape[2]+1))
+  lon1=np.zeros(lon.shape[0]+1)
+  dat1[:,:,:-1]=dat
+  dat1[:,:,-1]=dat[:,:,0]
+  lon1[:-1]=lon
+  lon1[-1]=lon[0]
+  return(dat1,lon1)
+
+#########################################
+# Input Data
+#########################################
+#parameters
+#g=9.81
+#CMIP6 models unit is m
+#does not need to separate by 9.81
+g=9.81
+#dlat=2.5
+ra=6371000.
+omega=7.2921/100000.
+
+#input file specified
+filein=os.environ["Z200_FILE"]; vname=z200_var
+print("Input File: "+filein)
+
+##Sometimes Pacific and Atlantic TUTTs are connected together
+##choose the separating longitude option here (Jay)
+##ridge_option=1, the script will search the longitude where the geopotential height is the largest between 120W-80W
+##the geopotential height is averaged between the subtropics (20-30N)
+##Sometimes the ridge may be weak over Central America (due to model biases), so the user can set perferred separating longitude (ridge_option=2)
+print("The script will search TUTT boundary based on zonal geostrophic wind (Ug) contour")
+UG_N15_target = float(input("Enter a Ug value (suggested between 1.0-2.0): "))
+
+print("Sometimes Pacific and Atlantic TUTTs are connected together")
+print("Please choose the separating longitude option here")
+print("ridge_option=1, the script will search the subtropical ridge over Central America (120W-80W)")
+print("ridge_option=2, the dividing longitude is specified by the user")
+ridge_option = int(input("Enter a number for ridge_option: "))
+if ridge_option==2:
+   user_dividing_longitude = int(input("Enter a number between 240-280 (120 W-80 W): "))
```

Comment:
The framework needs to run in a batch job environment, so unfortunately prompting the user for their input will not work here.